torch/_decomp/decompositions.py

```diff
@@ -1252,6 +1252,26 @@ def split_with_sizes(
     return splits
 
 
+# out_wrapper currently does not allow optional outputs
+@register_decomposition(
+    [aten.split_with_sizes_copy.default, aten.split_with_sizes_copy.out]
+)
+def split_with_sizes_copy(
+    self: Tensor,
+    split_sizes: List[int],
+    dim: int = 0,
+    out: Optional[List[Tensor]] = None,
+) -> Optional[List[Tensor]]:
+    splits = split_with_sizes(self, split_sizes, dim=dim)
+    if out is None:
+        return [s.clone() if s.is_contiguous() else s.contiguous() for s in splits]
```

Comment:
I think this is equivalent to just doing `s.clone()` irrespective of the contiguousness

Reply:
I thought so too but the `OpInfo` test on non-contiguous strided input reminded me otherwise:

```python
>>> foo = torch.zeros(5, 5).t()
>>> foo.clone().stride()
(1, 5)
>>> foo.contiguous().stride()
(5, 1)
```

Reply:
Ho in that case you want to force contiguous instead of "preserve" which is the default: `a.clone(memory_format=torch.contiguous_format)`